### PR TITLE
[pybullet] Add calculateJacobian.

### PR DESCRIPTION
**Pull request by bingjeff**
* Add the calculateJacobian method to the pybullet API.
* Adjust the shared memory interface to handle fixed/floating bases
  in the calculateJacobian method.
* Fix a few comments.

### CODE REVIEW

File: examples/pybullet/pybullet.c [hunk 1]
```diff
@@ -6947,6 +6953,147 @@ static PyObject* pybullet_calculateInverseDynamics(PyObject* self,
 	return Py_None;
 }
 
+/// Given an object id, joint positions, joint velocities and joint
+/// accelerations,
```
Comment (erwincoumans) [inline examples/pybullet/pybullet.c]: I guess the comment was copied from the Inverse Dynamics API?

File: examples/pybullet/pybullet.c [hunk 2]
```diff
@@ -6947,6 +6953,147 @@ static PyObject* pybullet_calculateInverseDynamics(PyObject* self,
...
+							double* linearJacobian = (double*)malloc(3 * byteSizeDofCount);
+							double* angularJacobian = (double*)malloc(3 * byteSizeDofCount);
+							b3GetStatusJacobian(statusHandle,
+												NULL,
+												linearJacobian,
+												angularJacobian);
+							if (linearJacobian)
+							{
+								PyObject* pymat = PyTuple_New(3);
+								for (int r = 0; r < 3; ++r) {
```
Comment (erwincoumans) [inline examples/pybullet/pybullet.c]: you can move the 'int r' out of the loop, to satisfy CI?

Comment (AndresTraks) [reply examples/pybullet/pybullet.c]: GCC 5 enables C99 features by default. Is it possible to upgrade to at least that instead?

Comment (erwincoumans) [reply examples/pybullet/pybullet.c]: This is Visual Studio compiling C code, simple fix.

**Pull request merged**